python2/httplib2/cookie.py

```diff
@@ -0,0 +1,402 @@
+            for path, cookies in domain_cookies.iteritems():
+                if not self._match_path(request_uri, path):
+                    continue
+
+                for cookie in cookies.itervalues():
+                    if cookie.is_expired():
+                        continue
+
+                    if cookie.secure and not is_https:
+                        continue
+
+                    cookie_header.append("{0}={1}".format(cookie.name, cookie.value))
+
+        return "; ".join(cookie_header)
+
+    @staticmethod
+    def parse_iter(header):
+        """An iterator to parse cookies in HTTP-style header(Set-Cookie).
+
+        cookie header example:
+            sid=vBj3zbrmgynsZ8ggVpmSIOc9bXAueT3z; HttpOnly;
+
+        It may contain multiple cookies separated by comma(RFC 2109).
+
+        Example:
+            BD_NOT_HTTPS=1; path=/; Max-Age=300, BIDUPSID=D2504FE922735DE;
+            expires=Thu, 31-Dec-37 23:55:55 GMT; max-age=2147483647; path=/;
+            domain=.example.com, PSTM=1595329370
+
+        The 'expires' also contains comma. Be careful!
+        """
+        if not header:
+            return
+
+        for cookie_str in RE_SPLIT_COOKIE.split(header):
+            attribute_list = cookie_str.split(";")
+            name, value = attribute_list[0].split("=")
+            cookie = {"name": name, "value": value}
+
+            for attr_str in attribute_list[1:]:
+                av_pair = attr_str.split("=", 1)
+                if len(av_pair) == 1:
+                    attr, val = av_pair[0], None
+                else:
+                    attr, val = av_pair
+
+                attr = attr.strip()
+                attr_l = attr.lower()
+
+                if attr_l == "max-age":
+                    cookie["max_age"] = val
+                elif attr_l == "expires":
+                    cookie["expires"] = val
+                elif attr_l == "domain":
+                    domain = val.lower()
+                    if not domain.startswith("."):
+                        domain = "." + domain
+                    cookie["domain"] = domain
+                elif attr_l == "path":
+                    cookie["path"] = val
+                elif attr_l == "secure":
+                    cookie["secure"] = True
+                elif val:
+                    # Other attrbute not handled
+                    cookie[attr] = val
+                else:
+                    # Other name-only attrbute not handled
+                    cookie[attr] = True
+
+            yield cookie
+
+    def extract_header(self, request_host, request_uri, header):
+        """Extract cookies from HTTP-style header(Set-Cookie) and save them
+        to cookie jar.
+
+        For an expired cookie, just save it and overwrite the old one to
+        disable the cookie.
+        """
+        for cookie in self.parse_iter(header):
+            domain = cookie.get("domain", request_host)
+            path = cookie.get("path", os.path.dirname(request_uri))
```

Comment:
(important)

This doesn't match `Cookie.normalize_path()` behavior. Are you sure it's ok they are different?

Reply:
These codes are for getting the cookie path from response. Here is the behavior when the `path` is omit in the `set-cookie`: https://tools.ietf.org/html/rfc6265#section-4.1.2.4

> If the server omits the Path attribute, the user
>    agent will use the "directory" of the request-uri's path component as
>    the default value.  (See Section 5.1.4 for more details.)

`Cookie.normalize_path()` only cares about how the path is stored in the Cookie. That's two different things.

Reply:
I was worried it would wrongly match like this:

```
set-cookie: name=value; path=/foo

GET /foobar HTTP/1.0
```
but tried and it seems to work as expected.